src/components/search/search.spec.tsx

```diff
@@ -312,25 +287,79 @@ describe("Search", () => {
     });
   });
 
-  describe("When button is true and textbox is active", () => {
-    it("does not render an icon in textbox", () => {
-      wrapper = renderSearch({
-        value: "",
-        searchButton: true,
-        id: "Search",
-        name: "Search",
-      });
-      const icon = wrapper
-        .find(Icon)
-        .findWhere((n) => n.props().type === "search")
-        .hostNodes();
-      act(() => {
-        const input = wrapper.find(Input);
-        input.simulate("focus");
-      });
-      wrapper.update();
-      expect(icon.props().value).toEqual(undefined);
+  it("should not render an icon in textbox when seachButton is passed a truthy boolean value", () => {
+    wrapper = renderSearch({
+      value: "",
+      searchButton: true,
+      id: "Search",
+      name: "Search",
+    });
+    const searchButton = wrapper.find(Button);
+    const icon = wrapper
+      .find(Icon)
+      .findWhere((n) => n.props().type === "search")
+      .hostNodes();
```

Comment:
**suggestion**: so this test doesn't return a false positive by accidentially finding the search button icon, what do you think of the following addition?:


```suggestion
    const inputIcon = wrapper
      .find(Input)
      .find(Icon)
      .findWhere((n) => n.props().type === "search")
      .hostNodes();
```

then you can do an existence assertion:

```ts
expect(inputIcon.exists()).toBeFalsy();
```

Reply:
Good idea, it'll need to be `.find(Input)` -> `.find(Textbox)` but it can definitely be scoped better

Reply:
also not sure asserting the props().value is correct here as it will always be undefined 🤦